packages/react-dom/src/__tests__/ReactErrorBoundaries-test.internal.js

```diff
@@ -799,45 +799,48 @@ describe('ReactErrorBoundaries', () => {
     expect(container.firstChild.textContent).toBe('Caught an error: Hello.');
   });
 
-  // @gate !disableModulePatternComponents
-  // @gate !disableLegacyContext
-  it('renders an error state if module-style context provider throws in componentWillMount', () => {
-    function BrokenComponentWillMountWithContext() {
-      return {
-        getChildContext() {
-          return {foo: 42};
-        },
-        render() {
-          return <div>{this.props.children}</div>;
-        },
-        UNSAFE_componentWillMount() {
-          throw new Error('Hello');
-        },
+  if (
+    !require('shared/ReactFeatureFlags').disableModulePatternComponents &&
+    !require('shared/ReactFeatureFlags').disableLegacyContext
+  ) {
+    it('renders an error state if module-style context provider throws in componentWillMount', () => {
```

Comment:
I'm not exactly sure what's going on with these tests that @gate doesn't work. I think it's that the test is expected to fail in prod, but doesn't because there's no deprecation warnings in prod.

Reply:
Needs to be:

```
// @gate !disableModulePatternComponents
// @gate !disableLegacyContext || !__DEV__
```

I fixed this in https://github.com/facebook/react/pull/28151